ui/app/components/identicon.js

```diff
@@ -16,16 +18,22 @@ function IdenticonComponent () {
   this.defaultDiameter = 46
 }
 
+function mapStateToProps (state) {
+  return {
+    useBlockie: state.metamask.useBlockie,
+  }
+}
+
 IdenticonComponent.prototype.render = function () {
   var props = this.props
-  const { className = '', address } = props
+  const { className = '', address, useBlockie } = props
   var diameter = props.diameter || this.defaultDiameter
 
   return address
     ? (
       h('div', {
         className: `${className} identicon`,
-        key: 'identicon-' + address,
+        key: useBlockie ? 'blockie' : 'identicon-' + address,
```

Comment:
`Key` needs to be unique to the element in case it's used in array, so including the address is a requirement, even if a blockie.

Reply:
Gotcha, fixing that now.